### PR TITLE
Fix Issue #21336: Defeault behavior for resolution property in Timestamps

### LINKED ISSUE #21336: Timestamp.resolution returns timedelta(microseconds=1)
I hadn't even noticed that `resolution` was a `datetime` property until now, but it returns `timedelta(0, 0, 1)`.  `Timestamp` doesn't override that.  I expect the correct thing to do is to return `Timedelta(nanoseconds=1)`.  Pretty low priority.

### PR DESCRIPTION
**Pull request by drewmassey**
- [x] closes #21336 
- [x] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

### CODE REVIEW

**Comment by mroeschke** [comment]
Looks good overall! Looks like the CI failed due to some linting errors.

File: doc/source/whatsnew/v0.23.1.txt
Comment (mroeschke) [inline doc/source/whatsnew/v0.23.1.txt]: I think the main point of the issue was that before the stated resolution was 1 microsecond while in actuality its 1 nanosecond (but we can't use `timedelta` to express that because it doesn't support nanoseconds)

**Comment by drewmassey** [comment]
@mroeschke This passed linting on my local machine, but now there is a simple conflict to resolve under what's new that I can't do because I don't have repo write access. Is this something you can resolve, or point me to someone who can? Thanks.

File: doc/source/whatsnew/v0.23.1.txt
Comment (drewmassey) [reply doc/source/whatsnew/v0.23.1.txt]: Resolved in commit c53f2c2

**Comment by mroeschke** [comment]
It's possible to fix the merge conflict locally and then re-push the resolution back to this branch. The general flow instructions can be found here: https://pandas.pydata.org/pandas-docs/stable/contributing.html#updating-your-pull-request

Once you execute the `git merge upstream/master` command in your console, you should be prompted to fix the merge conflict with the whatsnew document. Then you can commit the change and push back to this branch.

File: doc/source/whatsnew/v0.23.1.txt
Comment (mroeschke) [inline doc/source/whatsnew/v0.23.1.txt]: Looks like this already in the `**Other Fixes**` section.

**Comment by jbrockmendel** [comment]
LGTM.  Only nitpick would be with the wording "native pandas", but I don't have any better ideas.

**Review by mroeschke — APPROVED** [review]
LGTM. Is this appropriate for 0.23.1 @jreback?

**Comment by drewmassey** [comment]
Could I get a resolution on this from a contributor? Happy to fiddle the PR if necessary but don't want it to get stale.

File: doc/source/whatsnew/v0.23.1.txt
Comment (mroeschke) [inline doc/source/whatsnew/v0.23.1.txt]: Would you mind moving this entry to the `v0.23.2.txt` whatsnew? The 0.23.1 release is going out today or tomorrow I believe, and it appears the backports have already been established. Thanks for your patience.

**Comment by mroeschke** [comment]
Hmm looks like this branch picked up the commits of your other PR #21444. You'll need to revert the HEAD of this branch to a prior commit.

**Comment by jreback** [comment]
can you rebase

**Comment by mroeschke** [comment]
Looks like #21861 resolved this original issue (#21336). Would still greatly appreciate #21444!

**Pull request closed without merging**
